### PR TITLE
[DEPR] drop support for 'dashboard' and 'web-console' config option s…

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
…ections in favor of a new 'web-ui' section

# Description

part of https://github.com/inmanta/inmanta-ui/issues/317

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta_ui/config.py [hunk 1]
```diff
@@ -17,19 +17,26 @@
 """
 from inmanta.config import Option, is_bool, is_str
 
-web_console_enabled = Option("web-console", "enabled", True, "Should the server should host the web-console or not", is_bool)
+web_console_enabled = Option("web-ui", "enabled", True, "Whether the server should host the web-console or not", is_bool)
```
Comment (arnaudsjs) [inline src/inmanta_ui/config.py:20]: The idea behind the `web-ui` section is that it's generic and allows for more web-interfaces to be added in the future. From that perspective we probably don't want an `enabled` flag, but probably a `web_console_enabled` flag or something similar. @bartv Do you have a strong opinion here?

Comment (bartv) [reply src/inmanta_ui/config.py:20]: I would say console_enabled then.

We can even say: do we need an option to disable it?

**Comment by Hugo-Inmanta** [comment]
The tests on jenkins are expected to fail until https://github.com/inmanta/inmanta-core/pull/5046 has been merged in

File: src/inmanta_ui/config.py [hunk 2]
```diff
@@ -17,19 +17,30 @@
...
 )
+#############################
+# Dashboard
+#############################
+
+dash_realm = Option("web-ui", "dashboard_realm", "inmanta", "The realm to use for keycloak authentication.", is_str)
+dash_auth_url = Option("web-ui", "dashboard_auth_url", None, "The auth url of the keycloak server to use.", is_str)
+dash_client_id = Option(
+    "web-ui", "dashboard_client_id", None, "The client id configured in keycloak for this application.", is_str
+)
```
Comment (arnaudsjs) [inline src/inmanta_ui/config.py:46]: The dashboard was removed in ISO6, so it doesn't make sense to add it here.

File: changelogs/unreleased/317-add-web-ui-section-to-the-configuration-file-iso6.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+description: The 'dashboard' and 'web-console' sections are no longer supported for configuration options. The 'web-ui' section should now be used instead for configuration options related to web interfaces.
+change-type: major
+issue-nr: 317
+issue-repo: inmanta-ui
+destination-branches: [master]
+sections:
+  deprecation-note: "{{description}}"
```
Comment (arnaudsjs) [inline changelogs/unreleased/317-add-web-ui-section-to-the-configuration-file-iso6.yml:8]:
```suggestion
---
description: The 'web-console' section is no longer supported for configuration options. The 'web-ui' section should now be used instead for configuration options related to web interfaces.
change-type: major
issue-nr: 317
issue-repo: inmanta-ui
destination-branches: [master]
sections:
  deprecation-note: "{{description}}"
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta_ui/config.py [hunk 4]
```diff
@@ -17,19 +17,30 @@
...
 )
+#############################
+# Keycloak authentication
+#############################
+
+keycloak_realm = Option("web-ui", "keycloak_realm", "inmanta", "The realm to use for keycloak authentication.", is_str)
+keycloak_auth_url = Option("web-ui", "keycloak_auth_url", None, "The auth url of the keycloak server to use.", is_str)
+keycloak_client_id = Option(
+    "web-ui", "keycloak_client_id", None, "The client id configured in keycloak for this application.", is_str
+)
```
Comment (arnaudsjs) [inline src/inmanta_ui/config.py:46]: @bartv Do we need to prefix these options to make clear that they belong to the web_console?

Comment (bartv) [reply src/inmanta_ui/config.py:46]: We have never tested it but it should work with generic open id connect. So lets prefix them with with `oidc_`

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**